File: src/main/java/com/mathieucuvelier/customrunplugin/CustomRunConfigurationType.java
package com.mathieucuvelier.customrunplugin;

import com.intellij.execution.configurations.ConfigurationTypeBase;
import com.intellij.openapi.util.NotNullLazyValue;

public class CustomRunConfigurationType extends ConfigurationTypeBase {
    static final String ID = "CUSTOM_RUN_CONFIGURATION";

    public CustomRunConfigurationType() {
        super(ID, "Custom Run Configuration", "Run custom executables with configurable arguments", NotNullLazyValue.createValue(() -> PluginIcons.CUSTOM_RUN));
        addFactory(new CustomConfigurationFactory(this));
    }
}

File: src/main/kotlin/com/mathieucuvelier/customrunplugin/CustomRunConfigurationType.kt
package com.mathieucuvelier.customrunplugin

import com.intellij.execution.configurations.ConfigurationTypeBase
import com.intellij.openapi.util.NotNullFactory
import com.intellij.openapi.util.NotNullLazyValue
import javax.swing.Icon

class CustomRunConfigurationType : ConfigurationTypeBase(
    ID,
    "Custom Run Configuration",
    "Run custom executables with configurable arguments",
    NotNullLazyValue.createValue<Icon?>(
        NotNullFactory { PluginIcons.CUSTOM_RUN })
) {
    init {
        addFactory(CustomConfigurationFactory(this))
    }

    companion object {
        const val ID: String = "CUSTOM_RUN_CONFIGURATION"
    }
}